gap/matrix/classical.gi

```diff
@@ -759,26 +759,26 @@ RECOG.TestRandomElement := function (recognise, grp)
             Info(InfoClassical,2,o1, " ", o2);
             ### ACN March 2007 needed Projective order
             if  q mod 2 = 0 and (o1 <= 2 or o2 <= 2) then
-                return fail;
+                return TemporaryFailure;
             elif q mod 2 <> 0 and (o1 <= 4 or o2 <= 4) then
-                return fail;
+                return TemporaryFailure;
```

Comment:
@aniemeyer we need your input here... does the above change seem plausible. The old value `fail` / `NeverApplicable` means "give up, we have proven this can't be such-and-such group, while `TemporaryFailure` means it may be tried again later.

We couldn't see anything in your papers which would support that we abort here; and with is change, the non-generic SO(+1,4,q) get named much more reliably.

Reply:
I think false meant "we have proven that this can't be a classical group". So the lines where this PR actually changes behavior are the ones where false is replaced by temporary failure.

Reply:
Yeah I commented on the wrong line (once again proving that false/fail were a bad choice as return values